client/lib/products/constants.js

```diff
@@ -0,0 +1,6 @@
+// products constants
+export const PRODUCT_JETPACK_BACKUP = 'jetpack_backup';
```

Comment:
Seems like instead of introducing a new directory for those, we could use `lib/products-values/` and put `constants` there. That would make it easier to later get on the same line with what #36945 is suggesting in terms of using product slugs. 

But other than that, this file looks great!